tests/simulator/amr_convergence/compute_errors.py

```
@@ -0,0 +1,141 @@
+        for c, name in zip("xyz", ["mhdRhoVx", "mhdRhoVy", "mhdRhoVz"]):
+            f[f"rhoV{c}"] = rhoV[key][name]
+        f["Etot"] = etot[key]["mhdEtot"]
+        fields[key] = f
+    return fields
+
+
+def valid_mask(key, fine_box, ndim=2):
+    """1 where this patch's cells belong to the composite grid, 0 where a coarse
+    cell is covered by the fine level. fine_box = [lower, upper] in L0 cells."""
+    ilvl, lower, upper = key
+    shape = tuple(u - l + 1 for l, u in zip(lower, upper))
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)